dbt_platform_helper/utils/git.py

```diff
@@ -6,10 +6,11 @@ def git_remote():
     git_repo = subprocess.run(
         ["git", "remote", "get-url", "origin"], capture_output=True, text=True
     ).stdout.strip()
+    return extract_repository_name(git_repo)
 
-    if not git_repo:
-        return
 
-    _, repo = git_repo.split("@")[1].split(":")
+def extract_repository_name(repository_url):
+    if not repository_url:
+        return
 
-    return re.sub(r".git$", "", repo)
+    return re.search(r"([^/:]+?/[^/]+?)\.git", repository_url).group(1)
```

Comment:
I am OK with using a regular expression here, but because they are hard to understand, to help the next person and to validate if I have managed to understand it...
```suggestion
    # In the capturing group (the bit in the brackets)...
    # [^/:]+? Capture the organisation part (matches any character except : or / and captures as few characters as possible)
    # / Capture the slash after the organisation
    # [^/]+? Capture the repository name
    # After the capturing group...
    # Match, but don't capture the .git part at the end
    return re.search(r"([^/:]+?/[^/]+?)\.git", repository_url).group(1)
```

Reply:
Decided after discussion not to include the comment - would become out of date with the regex, makes the method significantly longer, and there are tools to help understand the regex (which we've made simpler)